### ISSUE TITLE
Unknown source amazon-ebs

### ISSUE THREAD

**Issue by tomkivlin** [hashicorp/packer#11402]
When filing a bug, please include the following headings if possible. Any
example text in this template can be deleted.

#### Overview of the Issue

Packer validation shows no problems, but when I run `packer build` I receive the following error:

```
1637235687,,ui,error,Error: Unknown source amazon-ebs.labs_webserver_ami\n\n  on  line 0:\n  (source code not available)\n\nKnown: []\n\n
```

#### Reproduction Steps

```
$ packer validate . 
The configuration is valid.
$
$ packer build -machine-readable packer-build.json.pkr.hcl
1637235687,,ui,error,Error: Unknown source amazon-ebs.labs_webserver_ami\n\n  on  line 0:\n  (source code not available)\n\nKnown: []\n\n
```

### Packer version

Packer v1.7.8

### Simplified Packer Template

```
data "amazon-ami" "latest_ubuntu" {
  filters = {
    name                = "ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-*"
    root-device-type    = "ebs"
    virtualization-type = "hvm"
  }
  most_recent = true
  owners      = ["099720109477"]
  region      = "${var.aws_region}"
}
locals {
    ami_name = "labs_webserver-${formatdate("YYYYMMDD-hhmm-Z", timestamp())}"
}
build {
  sources = ["source.amazon-ebs.labs_webserver_ami"]

  provisioner "ansible" {
    playbook_file = "../lab_webserver_ami.yml"
  }
}
source "amazon-ebs" "labs_webserver_ami" {
  ami_name                    = local.ami_name
  associate_public_ip_address = true
  communicator                = "ssh"
  force_deregister            = "true"
  instance_type               = "t3a.micro"
  pause_before_connecting     = "80s"
  region                      = "${var.aws_region}"
  run_tags = {
    Name = "packer-build-image"
  }
  source_ami   = "${data.amazon-ami.latest_ubuntu.id}"
  ssh_username = "${var.ssh_username}"
  subnet_id    = "${var.subnet_id}"
}
variable "aws_region" {
  type    = string
  default = "eu-west-3"
}

variable "ssh_username" {
  type    = string
  default = "ubuntu"
}

variable "subnet_id" {
  type    = string
  default = "subnet-1234....."
}

variable "vpc_id" {
  type    = string
  default = "vpc-1234...."
}
```

### Operating system and Environment details

Mac OS 11.6.1

### Log Fragments and crash.log files

```
$ PACKER_LOG=1 packer build -machine-readable packer-build.pkr.hcl | tee local-packer-output
2021/11/18 11:55:30 [INFO] Packer version: 1.7.8 [go1.17.2 darwin amd64]
2021/11/18 11:55:30 Old default config directory found: /Users/kivlint/.packer.d
2021/11/18 11:55:30 [TRACE] discovering plugins in /usr/local/bin
2021/11/18 11:55:30 Old default config directory found: /Users/kivlint/.packer.d
2021/11/18 11:55:30 [TRACE] discovering plugins in /Users/kivlint/.packer.d/plugins
2021/11/18 11:55:30 [TRACE] discovering plugins in .
2021/11/18 11:55:30 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2021/11/18 11:55:30 [INFO] PACKER_CONFIG env var set; attempting to open config file: /Users/kivlint/.packerconfig
2021/11/18 11:55:30 [WARN] Config file doesn't exist: /Users/kivlint/.packerconfig
2021/11/18 11:55:30 Old default config directory found: /Users/kivlint/.packer.d
2021/11/18 11:55:30 [INFO] Setting cache directory: /Users/kivlint/.cache/packer
2021/11/18 11:55:30 Old default config directory found: /Users/kivlint/.packer.d
2021/11/18 11:55:30 1637236530,,ui,error,Error: Unknown source amazon-ebs.labs_webserver_ami\n\n  on  line 0:\n  (source code not available)\n\nKnown: []\n\n
1637236530,,ui,error,Error: Unknown source amazon-ebs.labs_webserver_ami\n\n  on  line 0:\n  (source code not available)\n\nKnown: []\n\n
2021/11/18 11:55:30 [INFO] (telemetry) Finalizing.
2021/11/18 11:55:31 waiting for all plugin processes to complete...
```

**Comment by azr** [comment]
Hey @tomkivlin, thanks for opening, that's kind of odd, because this works on my machine. With or without manually requiring the `amazon` plugin. 

Also, this should print the available source in the `[]`, but it looks like none at all are available, super odd. Can you tell me a little bit more about your system, and how Packer was installed ? Thanks !

**Comment by tomkivlin** [comment]
When I raised the issue, packer was installed/updated by Homebrew and running 1.7.8.

I've just uninstalled that, and used the v1.7.8 binary download, same result.

I've then uninstalled that, and used the v1.7.7 binary download, same result.

Not sure if it makes a difference, but I was also running in a python3 virtual environment. I've tried outside that and still no joy.

I've also, just in case, run `packer init -upgrade .` and still same result.

But! (I include all the above and below for any coming across this in the future).

What appears to have made a difference is using the correct command....  I did have separate files (which I combined into the issue description, not thinking it might be important), as follows:

```
bootstrap/
├── data.pkr.hcl
├── locals.pkr.hcl
├── build.pkr.hcl
├── sources.pkr.hcl
├── variables.pkr.hcl
```

So, changing my command to reference the directory rather than the build file, it works:
`packer build .`

Explains why `packer validate .` worked but `packer build build.pkr.hcl` didn't.
